src/openrct2/world/Footpath.cpp

```diff
@@ -114,10 +114,10 @@ static bool entrance_has_direction(TileElement* tileElement, int32_t direction)
 TileElement* map_get_footpath_element(int32_t x, int32_t y, int32_t z)
 {
     TileElement* tileElement = map_get_first_element_at(x, y);
+    if (tileElement == nullptr)
```

Comment:
revert. Previous code works fine and tends to be the standard for `map_get...`

Reply:
But it will crash on:
```cpp
        if (tileElement->GetType() == TILE_ELEMENT_TYPE_PATH && tileElement->base_height == z)
```
If `titleElement` is ever `nullptr`. Should we assert instead?

Reply:
It will leave the loop before that due to the break.

Reply:
I reverted, but am not sure I follow, there is no break within the function and thus it will go into `do while` and try to dereference `tileElement`, which likely will crash if it's a `nullptr`

Reply:
Ah my mistake I meant revert both of your changes you made. It used to have a break on nullptr in the do while.

Reply:
So do I keep them as is or do I move them (this and the open issue below) to within the `do...while` and `break` instead of returning?

Reply:
I want them to be within the `do while` with a `break`. This will then match the format of other `map_get...`